File: Abstract-Factory/Python/light_theme_factory.py
from ui_factory import UIFactory
from light_button import LightButton
from light_checkbox import LightCheckbox

class LightThemeFactory(UIFactory):
    def create_button(self) -> LightButton:
        return LightButton()

    def create_checkbox(self) -> LightCheckbox:
        return LightCheckbox()

File: Abstract-Factory/PHP/LightThemeFactory.php
<?php

declare(strict_types=1);

require_once __DIR__ . '/UIFactory.php';
require_once __DIR__ . '/LightButton.php';
require_once __DIR__ . '/LightCheckbox.php';

class LightThemeFactory implements UIFactory {
    public function createButton(): Button {
        return new LightButton();
    }

    public function createCheckbox(): Checkbox {
        return new LightCheckbox();
    }
}
